operator/pkg/gateway-api/httproute.go

```diff
@@ -41,25 +42,56 @@ func newHTTPRouteReconciler(mgr ctrl.Manager) *httpRouteReconciler {
 func (r *httpRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &gatewayv1.HTTPRoute{}, backendServiceIndex,
 		func(rawObj client.Object) []string {
-			hr, ok := rawObj.(*gatewayv1.HTTPRoute)
+			route, ok := rawObj.(*gatewayv1.HTTPRoute)
 			if !ok {
 				return nil
 			}
 			var backendServices []string
+			for _, rule := range route.Spec.Rules {
+				for _, backend := range rule.BackendRefs {
+					namespace := helpers.NamespaceDerefOr(backend.Namespace, route.Namespace)
+					backendServiceName, err := helpers.GetBackendServiceName(r.Client, namespace, backend.BackendObjectReference)
+					if err != nil {
+						log.WithFields(logrus.Fields{
+							logfields.Controller: "httpRoute",
+							logfields.Resource:   client.ObjectKeyFromObject(rawObj),
+						}).WithError(err).Error("Failed to get backend service name")
```

Comment:
in case of an error, we need to `continue` without adding the service to the slice

Reply:
Oups indeed , good catch :+1: